src/python/pants/backend/python/rules/run_python_binary.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.backend.python.rules.create_python_binary import PythonBinaryFieldSet
+from pants.backend.python.rules.pex import Pex, PexPlatforms
+from pants.backend.python.rules.pex_from_targets import PexFromTargetsRequest
+from pants.backend.python.rules.python_sources import (
+    UnstrippedPythonSources,
+    UnstrippedPythonSourcesRequest,
+)
+from pants.backend.python.target_types import PythonBinaryDefaults, PythonBinarySources
+from pants.core.goals.binary import BinaryFieldSet
+from pants.core.goals.run import RunRequest
+from pants.core.util_rules.determine_source_files import AllSourceFilesRequest, SourceFiles
+from pants.engine.addresses import Addresses
+from pants.engine.fs import Digest, MergeDigests
+from pants.engine.rules import SubsystemRule, rule
+from pants.engine.selectors import Get, MultiGet
+from pants.engine.target import InvalidFieldException, TransitiveTargets
+from pants.engine.unions import UnionRule
+
+
+@rule
+async def run_python_binary(
+    field_set: PythonBinaryFieldSet, python_binary_defaults: PythonBinaryDefaults
+) -> RunRequest:
```

Comment:
It feels like rather than being binary (or repl) specific, this is connected to `TwoStepPex`... where rather than creating two pexes (one for requirements, the other for sources, iirc) we instead could use a requirements pex, and a `Digest` of loose sources.

`Repl` already uses `TwoStepPex`, so perhaps starting there would make sense?

Reply:
I don't think so. We use `TwoStepPex` for `binary` and `awslambda`, actually. `TwoStepPex` results in one single Pex file. I think it's important that it continues to behave the same way.

I plan to change `repl` tomorrow to behave the same way as `run`. If it helps, I can bundle it into this PR. Only wanted to get this one up as a checkpoint and to make it in time for the release.

Reply:
> I don't think so. We use TwoStepPex for binary and awslambda, actually. TwoStepPex results in one single Pex file. I think it's important that it continues to behave the same way.

Got it: my mistake.

> I plan to change repl tomorrow to behave the same way as run. If it helps, I can bundle it into this PR. Only wanted to get this one up as a checkpoint and to make it in time for the release.

No need to bundle. Thanks!

Reply:
No mistake - it was a good question! I was wondering around the same time you wrote the original question "Hm, can we delete TwoStepPex now?"